File: gasolinaRubi/src/main/java/com/corral/mityc/SplashActivity.java
package com.corral.mityc;

import android.app.Activity;
import android.content.Intent;
import android.os.AsyncTask;
import android.os.Bundle;

/**
 * Splash Screen
 */
public class SplashActivity extends Activity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.splash_screen);
        startHeavyProcessing();

    }

    private void startHeavyProcessing(){
        new LongOperation().execute("");
    }

    private class LongOperation extends AsyncTask<String, Void, String> {

        @Override
        protected String doInBackground(String... params) {

            for (int i = 0; i < 5; i++) {
                try {
                    Thread.sleep(500);
                } catch (InterruptedException e) {
                    Thread.interrupted();
                }
            }
            return "";
        }

        @Override
        protected void onPostExecute(String result) {
            Intent i = new Intent(SplashActivity.this, MitycRubi.class);
            startActivity(i);
            finish();
        }

        @Override
        protected void onPreExecute() {}

        @Override
        protected void onProgressUpdate(Void... values) {}
    }
}

File: gasolinaRubi/src/main/java/com/corral/mityc/SplashActivity.kt
package com.corral.mityc

import android.app.Activity
import android.content.Intent
import android.os.Bundle

class SplashActivity : Activity() {


        override fun onCreate(icicle : Bundle?) {
            super.onCreate(icicle)

            /*
             * Damos valor a Constantes.DEBUG
             * Nos será útil para la clase MLog, donde mostramos log, sólo si
             * estamos depurando código.
             */
            val compileMode = getString(R.string.COMPILE_MODE)
            Constantes.DEBUG = !compileMode.equals("release", ignoreCase = true)

            val intent = Intent(this, MitycRubi::class.java)
            startActivity (intent)
            finish()
        }
}